test-helper/src/lib.rs

```diff
@@ -99,3 +108,305 @@ pub fn generate_plan_tests(_tokens: TokenStream) -> TokenStream {
         .collect::<proc_macro2::TokenStream>()
         .into()
 }
+
+#[proc_macro]
+pub fn generate_docker_tests(_tokens: TokenStream) -> TokenStream {
```

Comment:
I am unsure about this change. I think it makes editing how the tests are run a lot less clear and means we can't easily add custom behaviour for a single test if we want to in the future. Each example also needs to be manually edited anyway to account for the test specific variables in `test.env`, which isn't that different then just editing `docker_run_tests`.

Reply:
Its to remove code duplication, most tests are technically the same anyway, just different paths and outputs. Especially now with the `test.env` file is there, new contributors can add tests easily to the examples.

Reply:
If you want to, I can add another section to the Contributing Guide on how to add tests.

Reply:
I don't think it removes code duplication that much though. The tests currently are only a few lines long
```ts
#[test]
fn test_node_custom_version() {
    let name = simple_build("./examples/node-custom-version");
    let output = run_image(name, None);
    assert!(output.contains("Node version: v18"));
}
```

Moving that to a `test.env` file means that the test inputs are now separated across many different files. For the generate plan tests it makes sense because we just run a single function on all the examples. But the docker run tests have a lot more inputs and possibilities. Editing the docker container/run logic is less trivial without prior knowledge of how this works. It also means we cannot use the same example directory for multiple tests, which we do quite a bit.